packages/common/src/constraints/Values.ts

```diff
@@ -116,7 +116,10 @@ export const getLocationValues = async (
   const locationMap = new Map<string, number>();
   const locationArr: { name: string; numberOfNodes: number }[] = [];
   for (const candidate of validCandidates) {
-    const candidateLocation = await getCandidateLocation(candidate.name);
+    const candidateLocation = await getCandidateLocation(
```

Comment:
a candidate is uniquely identified by its slot id, we should be consistent with that.

Could you retrieve the candidate's slotId by name and stash and then fetch the location by slotId ?